indigo_app/views/works.py

```diff
@@ -556,131 +558,58 @@ def get_context_data(self, **kwargs):
     def form_valid(self, form):
         error = None
         works = None
+        locality_code = self.locality.code if self.locality else None
+        bulk_creator = plugins.for_locale('bulk-creator', self.country.code, None, locality_code)
 
         try:
             table = self.get_table(form.cleaned_data.get('spreadsheet_url'))
-            works = self.get_works(table, form)
-            self.link_works(works, form)
+            works = bulk_creator.get_works(self, table)
+            self.create_links(works, form)
```

Comment:
I suspect it'll be useful to have `create_links` and `get_tasks` both exist on the plugin. Time will tell.